src/runtime_src/core/common/xclbin_parser.cpp

```diff
@@ -850,6 +850,30 @@ get_softkernels(const axlf* top)
   return sks;
 }
 
+std::vector<aieresourcesbin_object>
+get_aieresourcesbin(const axlf* top)
+{
+  std::vector<aieresourcesbin_object> vec;
+
+  for (auto section = ::xclbin::get_axlf_section(top, AIE_RESOURCES_BIN);
+    section != nullptr;
+    section = ::xclbin::get_axlf_section_next(top, section, AIE_RESOURCES_BIN)) {
+      auto begin = reinterpret_cast<const char*>(top) + section->m_sectionOffset;
+      auto aieres = reinterpret_cast<const aie_resources_bin*>(begin);
```

Comment:
Perhaps we should have a function to do this to hide these casts if we do this often for different kinds of objects.